src/microsoft/office/license/user/microsoft-office-license-user.controller.js

```diff
@@ -1,73 +1,50 @@
-            users: false,
-            userPoll: {}
-        };
-
-        this.$scope.$on("microsoft.office.license.user.add", this.delayedGetUser);
-        this.$scope.$on("microsoft.office.license.user.edit", this.delayedGetUser);
-        this.$scope.$on("microsoft.office.license.user.delete", this.delayedGetUser);
+        this.$scope.$on("microsoft.office.license.user.add", () => this.delayedGetUsers());
+        this.$scope.$on("microsoft.office.license.user.edit", () => this.delayedGetUsers());
+        this.$scope.$on("microsoft.office.license.user.delete", () => this.delayedGetUsers());
 
-        this.getUsersId();
+        this.getUserIds();
     }
 
-    transformItem (item) {
-        return this.license.getUserDetails(this.$scope.currentLicense, item)
+    transformItem ({ id }) {
+        return this.license.getUserDetails(this.$scope.currentLicense, id)
             .then((details) => {
                 if (details.status !== "ok") {
-                    this.loaders.userPoll[item] = true;
-
-                    this.license.pollUserDetails(this.$scope.currentLicense, item, this.$scope)
-                        .then(() => {
-                            this.loaders.userPoll.item = false;
-                            return this.delayedGetUser();
-                        })
-                        .catch((err) => {
-                            this.loaders.userPoll.item = false;
-                            return err;
-                        });
+                    details.isLoading = true;
+                    this.license.pollUserDetails(this.$scope.currentLicense, id, this.$scope)
+                        .then(() => this.delayedGetUsers())
+                        .finally(() => { details.isLoading = false; });
                 }
                 return details;
             })
             .catch((err) => {
-                this.loaders.userPoll.item = false;
-                return err;
+                this.Alerter.error(this.translator.tr("microsoft_office_license_detail_user_error") + err.message, this.$scope.alerts.main);
+                return { id, activationEmail: id };
             });
     }
 
-    onTransformItemDone () {
-        this.loaders.users = false;
-    }
-
 
-    getUsersId () {
-        this.loaders.users = true;
+    getUserIds () {
         this.users = null;
 
         return this.license.getUsers(this.currentLicense)
-            .then((users) => { this.users = users; })
-            .catch((err) => this.Alerter.error(err.message, this.$scope.alerts.tabs))
-            .finally(() => {
-                if (_.isEmpty(this.users)) {
-                    this.loaders.users = false;
-                }
-            });
+            .then((users) => { this.users = users.map((id) => ({ id })); })
+            .catch((err) => this.Alerter.error(err.message, this.$scope.alerts.tabs));
+    }
+
+    delayedGetUsers () {
+        return this.$timeout(() => this.getUserIds(), 250);
```

Comment:
Why is this delayed ?

Reply:
I don't really know because it was already like that before I made some changes but what I know is, if there's no delay it fails sometime !
But I haven't investigate a lot on it !